lib/house.rb

```diff
@@ -0,0 +1,41 @@
+class House
+  attr_reader :price,
+              :address,
+              :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  # def price
+  #   @price.slice(0).to_i
+  # end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    return true if @price.to_i > 500000
+    false
+  end
+
+  def rooms_from_category(category)
+    @rooms.find_all do |room|
+      room.category == category
+    end
+  end
+
+  def area
```

Comment:
This is close, you would need another variable you set to 0 outside of your each. So
```
house_area = 0

@rooms.each do |room|
   house_area = room.length * room.width
 end

house_area
```